### ISSUE TITLE
MapSnapshot not generating bitmap

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#10351]
I occasionally have issues rendering a MapSnapshot. After minutes of waiting the SnapshotReadyCallback/ErrorHandler hasn't been invoked. Haven't found a reproducible way, debugging always results in invoking the callback. 

![ezgif com-video-to-gif 2](https://user-images.githubusercontent.com/2151639/32300663-cfcda264-bf17-11e7-9992-c09a5fd91075.gif)

Difference in logs for a succes vs fail:

Success:

```
I/MapSnapshotterMarkerAct: Starting snapshot
V/HTTPRequest: [HTTP] Request was successful (code = 200).
V/HTTPRequest: [HTTP] Request was successful (code = 200).
V/HTTPRequest: [HTTP] Request was successful (code = 200).
V/HTTPRequest: [HTTP] Request was successful (code = 200).
I/MapSnapshotterMarkerAct: Snapshot ready
```

Fail:

```
I/MapSnapshotterMarkerAct: Starting snapshot
V/HTTPRequest: [HTTP] Request was successful (code = 200).
```

**Comment by tobrun** [comment]
Also seeing occurences where I'm not seeing any http request coming in.

**Comment by Guardiola31337** [comment]
I can reproduce OP in all the activities that use the map snapshotter with a Nexus 5.

```
11-08 13:53:42.822 14657-14657/com.mapbox.mapboxsdk.testapp I/MapSnapshotterMarkerAct: Starting snapshot
11-08 13:53:42.836 14657-21903/com.mapbox.mapboxsdk.testapp W/mbgl: {Map Snapshotter}[OpenGL]: Not using Vertex Array Objects
11-08 13:53:42.849 776-806/? I/ActivityManager: Displayed com.mapbox.mapboxsdk.testapp/.activity.snapshot.MapSnapshotterMarkerActivity: +119ms
```

The Nexus 5 uses an Adreno 330 👉 https://github.com/mapbox/mapbox-gl-native/blob/13ec2cc562ce98a089b47dee2987b5e58a0384e2/src/mbgl/gl/context.cpp#L252

cc @tobrun @ivovandongen

**Comment by ivovandongen** [comment]
> I can reproduce OP in all the activities that use the map snapshotter with a Nexus 5.

@Guardiola31337 Are you sure it is in all the activities or just the marker example like in the OP?

**Comment by Guardiola31337** [comment]
Retest in branch `ivd-fix-mapsnapshot-ref-issue` (#10418) and now all the activities are working.
BTW, I'm still seeing `{Map Snapshotter}[OpenGL]: Not using Vertex Array Objects` log messages. Not sure if related.

**Comment by ivovandongen** [comment]
> BTW, I'm still seeing {Map Snapshotter}[OpenGL]: Not using Vertex Array Objects log messages. Not sure if related.

See https://github.com/mapbox/mapbox-gl-native/pull/10253
